### PR TITLE
Replace window.confirms with the replay-confirm dialog

### LINKED ISSUE #4374: Switch window.confirms to ConfirmDialog
Related to https://github.com/RecordReplay/devtools/issues/3993

I figured this was pretty low-effort and worth following up on.
Switch the remaining window.confirms over to the new confirm dialog.

- [x] WorkspaceMember https://github.com/RecordReplay/devtools/blob/65f2d3dd027e182e3d29916d356e7a81c1485284/src/ui/components/shared/WorkspaceSettingsModal/WorkspaceMember.tsx#L143
- [x] PendingTeamPrompt https://github.com/RecordReplay/devtools/blob/c1087a9e67ec1fb101e04bfd872c6766595cabaa/src/ui/components/Library/PendingTeamPrompt.tsx#L27
- [x] RecordingOptionsDropdown https://github.com/RecordReplay/devtools/blob/a0c76b5d2ee4265f8eb7c3f87d106770cadca546/src/ui/components/Library/RecordingOptionsDropdown.tsx#L42
- [x] BatchActionDropdown https://github.com/RecordReplay/devtools/blob/a0c76b5d2ee4265f8eb7c3f87d106770cadca546/src/ui/components/Library/BatchActionDropdown.tsx#L40
- [x] APIKeys https://github.com/RecordReplay/devtools/blob/a0c76b5d2ee4265f8eb7c3f87d106770cadca546/src/ui/components/shared/APIKeys.tsx#L83
- [x] NewWorkspaceModal https://github.com/RecordReplay/devtools/blob/6ffea84e53ad3f441c3faae08f8bea47e9512192/src/ui/components/shared/NewWorkspaceModal/NewWorkspaceModal.tsx#L257

### PR DESCRIPTION
**Pull request by colbyr**
Fixes #4374 

Took a few minutes to replace the rest of our window.confirms with the new confirmation dialog!

For example...
Screen Shot 2021-11-11 at 11 18 52 AM

Very open to feedback on the copy here!

### CODE REVIEW

File: src/ui/components/Library/BatchActionDropdown.tsx [hunk 1]
```diff
@@ -26,23 +42,20 @@ function BatchActionDropdown({
...
-      count == 1 ? "" : "s"
-    }. \n\nAre you sure you want to proceed?`;
-
-    if (window.confirm(message)) {
-      selectedIds.forEach(recordingId => deleteRecording(recordingId, currentWorkspaceId));
-      setSelectedIds([]);
-    }
-
-    setExpanded(false);
+    confirmDestructive(getConfirmOptions(selectedIds.length)).then(confirmed => {
```
Comment (colbyr) [inline src/ui/components/Library/BatchActionDropdown.tsx:52]: Screen Shot 2021-11-11 at 11 17 09 AM

File: src/ui/components/Library/PendingTeamPrompt.tsx [hunk 2]
```diff
@@ -18,16 +19,22 @@ function PendingTeamPrompt({ workspace, setWorkspaceId }: PendingTeamPromptProps
...
   const handleDecline = () => {
-    const message = "Are you sure you want to decline this invitation?";
-    if (window.confirm(message)) {
-      declinePendingInvitation({ variables: { workspaceId: id } });
-      setLoading(true);
-    }
+    confirmDestructive({
+      message: "Are you sure you want to decline this invitation?",
+      acceptLabel: "Decline invitation",
+    }).then(confirmed => {
```
Comment (colbyr) [inline src/ui/components/Library/PendingTeamPrompt.tsx:32]: Screen Shot 2021-11-11 at 11 20 55 AM

File: src/ui/components/Library/RecordingOptionsDropdown.tsx [hunk 3]
```diff
@@ -29,20 +30,25 @@ function RecordingOptionsDropdown({
...
-    if (window.confirm(message)) {
-      deleteRecording(recordingId, currentWorkspaceId);
-    }
-    setExpanded(false);
+    confirmDestructive({
+      message: "Delete replay?",
+      description:
+        "This action will permanently delete this replay. \n\nAre you sure you want to proceed?",
+      acceptLabel: "Delete replay",
+    }).then(confirmed => {
```
Comment (colbyr) [inline src/ui/components/Library/RecordingOptionsDropdown.tsx:46]: Screen Shot 2021-11-11 at 11 22 52 AM

File: src/ui/components/shared/APIKeys.tsx [hunk 4]
```diff
@@ -77,12 +79,16 @@ function ApiKeyList({ apiKeys, onDelete }: { apiKeys: ApiKey[]; onDelete: (id: s
...
-
-                  if (window.confirm(message)) {
-                    onDelete(apiKey.id);
-                  }
+                  confirmDestructive({
+                    message: "Delete API key?",
+                    description:
+                      "This action will permanently delete this API key. \n\nAre you sure you want to proceed?",
+                    acceptLabel: "Delete API key",
+                  }).then(confirmed => {
```
Comment (colbyr) [inline src/ui/components/shared/APIKeys.tsx:87]: Screen Shot 2021-11-11 at 11 23 42 AM

File: src/ui/components/shared/NewWorkspaceModal/NewWorkspaceModal.tsx [hunk 5]
```diff
@@ -251,10 +255,12 @@ function SlideBody2({ hideModal, setCurrent, newWorkspace, total, current }: Sli
...
   const didUserConfirm = () => {
-    const message =
-      "You started tried to invite someone, but didn't press the invite button. Are you sure you want to proceed?";
-
-    return window.confirm(message);
+    return confirmDestructive({
+      message: "Continue without sending invitations?",
+      description:
+        "You started to invite someone, but didn't press the invite button. Are you sure you want to proceed?",
+      acceptLabel: "Continue anyways",
```
Comment (colbyr) [inline src/ui/components/shared/NewWorkspaceModal/NewWorkspaceModal.tsx:262]: Screen Shot 2021-11-11 at 11 18 52 AM

File: src/ui/components/shared/WorkspaceSettingsModal/WorkspaceMember.tsx [hunk 6]
```diff
@@ -135,14 +136,19 @@ export function NonRegisteredWorkspaceMember({
...
     setExpanded(false);
-    const message = `Are you sure you want to remove ${member.email} from this team?`;
-
-    if (window.confirm(message)) {
-      deleteUserFromWorkspace({ variables: { membershipId: member.membershipId } });
-    }
+    confirmDestructive({
+      message: "Remove team member?",
+      description: `Are you sure you want to remove ${member.email} from this team?`,
+      acceptLabel: "Remove them",
```
Comment (colbyr) [inline src/ui/components/shared/WorkspaceSettingsModal/WorkspaceMember.tsx:146]: Screen Shot 2021-11-11 at 11 24 46 AM

**Review by jcmorrow — APPROVED** [review]
I really love the design of this modal. Looks great 🧲

**Comment by jcmorrow** [comment]
Also, beautiful job on the documentation both in this PR and in the linked issue @colbyr! 👏

File: src/ui/components/shared/Button.tsx [hunk 7]
```diff
@@ -69,7 +69,10 @@ function getColorClasses(color: Colors, style: ButtonStyles) {
 export function getButtonClasses(color: Colors, style: ButtonStyles, size: ButtonSizes) {
   const standardClasses = STANDARD_CLASSES[size];
   const colorClasses = getColorClasses(color, style);
-  const focusClasses = "focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-blue-500";
+  const focusClasses = `focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-${getColorCode(
+    color,
```
Comment (colbyr) [inline src/ui/components/shared/Button.tsx:73]: Changed the color of the focus ring to match the button's color per Jon's request.

Screen Shot 2021-11-11 at 1 21 26 PM

**Comment by colbyr** [comment]
Thanks @jcmorrow 😎

**Pull request merged**